### PR TITLE
test: add test cases for defineRouter

### PR DESCRIPTION
**Pull request by himself65**
Cover case: https://github.com/dai-shi/waku/issues/536

### CODE REVIEW

File: e2e/fixtures/waku-router/src/entries.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    component: AboutPage,
+  });
+
+  createPage({
+    render: 'dynamic',
+    path: '/[roomId]',
+    component: RoomPage,
+  });
+  // fixme: error TS2742
+}) as any;
```
Comment (himself65) [inline e2e/fixtures/waku-router/src/entries.tsx:33]: IDK

File: tsconfig.json [hunk 2]
```diff
@@ -16,11 +16,7 @@
     "declaration": true,
     "emitDeclarationOnly": true,
     "composite": true,
-    "incremental": true,
-    "paths": {
-      "waku": ["./packages/waku/src/main.ts"],
-      "waku/*": ["./packages/waku/src/*.ts"]
-    }
```
Comment (himself65) [inline tsconfig.json:23]: I found that examples are aliases to the src file, I think this is not good. we should type check from output

File: e2e/fixtures/waku-router/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "waku-router",
+  "version": "0.1.0",
+  "type": "module",
+  "private": true,
+  "scripts": {
+    "dev": "waku dev",
+    "build": "waku build",
+    "start": "waku start"
```
Comment (dai-shi) [inline e2e/fixtures/waku-router/package.json:9]: Is it intentional to not add `--with-ssr`?

Comment (himself65) [reply e2e/fixtures/waku-router/package.json:9]: It's okay for here since test case will run both cases

**Comment by himself65** [comment]
```
 Locator: getByTestId('current-url')
    Expected string: "/"
    Received string: ""
```

It works perfectly on my PC, let me see tmr

File: e2e/fixtures/waku-router/package.json (hunk 3 above)
Comment (dai-shi) [reply e2e/fixtures/waku-router/package.json:9]: I see.

**Pull request closed without merging**